clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2056,13 +2056,28 @@ void validateAndRetrieveValueSchemas(
     if (replicationSchemasResponse.isError()) {
       LOGGER.error("Failed to fetch replication metadata schemas!" + replicationSchemasResponse.getError());
     } else {
-      // We only need a single valid schema, so getting the first one is good enough.
       if (replicationSchemasResponse.getSchemas() != null && replicationSchemasResponse.getSchemas().length > 0) {
-        pushJobSetting.replicationMetadataSchemaString = replicationSchemasResponse.getSchemas()[0].getSchemaStr();
-        LOGGER.info(
-            "Retrieved and using schema with id: {}, and string: {}",
-            replicationSchemasResponse.getSchemas()[0].getId(),
-            pushJobSetting.replicationMetadataSchemaString);
+        boolean foundRmdSchema = false;
+        for (MultiSchemaResponse.Schema schema: replicationSchemasResponse.getSchemas()) {
+          if (schema.getRmdValueSchemaId() == pushJobSetting.valueSchemaId
+              && schema.getId() > pushJobSetting.rmdSchemaId) {
+            pushJobSetting.rmdSchemaId = schema.getId();
+            pushJobSetting.replicationMetadataSchemaString = schema.getSchemaStr();
+            foundRmdSchema = true;
+          }
+        }
```

Comment:
I'm not sure this is the right way to get the right rmd schema id for the store. We need to use the rmd schema id that is set at the store config as that is the one which the new version will use, and the one that servers are expecting.

If that is not feasible, then we need to specify a way to "override" the version level config for RMD schema id

Reply:
Also, it's possible I might have missed it, but we should also verify if the schema that the user specified in the input matches the RMD schema that this code is deducing.

We probably need to specify some "conversion logic" as well - otherwise things will break when RMD schema ever evolves

Reply:
We support the long to RMD conversion in oSS and added a check for schema subset check otherwise. Extensions of VPJ (internally at LI) will be handled as the conversion is specific to LI online -> offline contracts. We can extract it to OSS if we find value in supporting the general upconversion of RMD